content/blog/rust-production-guide/index.md

```diff
@@ -0,0 +1,351 @@
+   - Costs of delaying other projects and features. / Time spent on Rust could be spent on other projects.
+   
+3. **Training and Team Preparation**
+   - Training (4 months of dedicated time)
+   - Rust training and resources (books, online courses, conferences)
+   - Mentorship and coaching
+   - Team building activities (team lunches, offsites)
+
+4. **Hiring and Team Expansion**
+   - Staffing costs for Rust developers (recruiting, onboarding)
+   - Hiring people in infrastructure roles (SREs, DevOps)
+
+5. **Setup Costs**
+   - Project structure
+   - Workflows and tools (IDEs, linters, formatters)
+   - CI/CD pipelines
+
+6. **Development and Implementation**
+   - Costs of fast development machines ([Rust builds are hardware-intensive](/blog/tips-for-faster-rust-compile-times/))
+   - Development costs (coding, testing, debugging)
+   - Documentation
+   - Code reviews and pair programming (to ensure code quality and knowledge sharing)
+   - Performance tuning (if needed)
+   - Security audits
+   - Compliance and legal reviews (regulations and standards, e.g. GDPR or software licenses)
+
+7. **Infrastructure and Tooling**
+   - Licensing for proprietary tools (rarely needed, but consider it)
+   - Additional hardware during migration (a separate set of infrastructure for the new system)
+
+8. **Ongoing Maintenance and Support**
+   - Cloud costs (compute, storage, networking)
+   - Ongoing maintenance costs (monitoring, logging, on-call, CI/CD pipelines)
+   - Refactoring and cleaning up technical debt
+   
+9.  **Communication and Marketing**
+    - Internal communication (presentations)
+    - External communication (blog posts, conference talks)
+
+10. **Risk Management**
+    - Account for the cost of failure
+    - Consider costs of rollback mechanisms
+    - Costs of rollback if the migration fails
+
+## On Hiring Talent
+
+Hiring Rust talent takes time and effort.
+This section provides insights into hiring strategies and what to look for in candidates for Rust projects specifically.
+
+### People with Production Experience Are Hard to Find and Expensive
+
+It's challenging to find developers with Rust *production* experience. Competing with the crypto industry, which heavily uses Rust, makes it even harder. Salaries for Rust developers in crypto can be 2-3 times higher than in other industries, and the pool of talent is small.
+
+Unless you are a hot new startup or have a big budget, finding people with Rust production experience can be challenging and costly. Consider training your own team instead.
+
+### Source Talent From Your Existing Teams Instead
+
+Instead of focusing solely on experience, consider finding junior- to mid-level developers who are smart, curious, and eager to learn. From my experience, good candidates often have experience in related areas (like Kotlin or TypeScript) and are known for being quick learners. Place them near your Rust champion or an expert mentor to help them get up to speed quickly.
```

Comment:
junior to mid-level?

Extra hyphen to junior?

Reply:
Yes, there should be a hyphen after "junior" when used as a compound adjective before "to mid-level developers." 😉